### PR TITLE
Moving attention, GPTBlocks to their own folder

### PR DESCRIPTION
**Pull request by bcui19**
As PR name. Trying to make the file structure a little cleaner.

### CODE REVIEW

File: examples/llm/src/modules/__init__.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+
+__all__ = [
+    'TorchCausalAttention',
+    'FlashCausalAttention',
+    'TritonFlashCausalAttention',
+    'GPTMLP',
+    'GPTBlock',
+]
```
Comment (vchiley) [inline examples/llm/src/modules/__init__.py:11]:
```suggestion
]

```

add newline

File: examples/llm/src/modules/attention_modules.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+        alibi_bias = alibi_bias - torch.arange(
+            1 - seq_len, 1, dtype=dtype, device=device).view(1, 1, seq_len, 1)
+        alibi_bias.abs_().mul_(
+            -1
+        )  # since we're using causal flag, this isn't really needed, but why not include it
+
+    m = torch.arange(1, n_heads + 1, dtype=dtype, device=device)
+    m.mul_(alibi_bias_max / n_heads)
+    alibi_bias = alibi_bias * (1. / (2**m.view(1, n_heads, 1, 1)))
+    return alibi_bias
```
Comment (vchiley) [inline examples/llm/src/modules/attention_modules.py:241]:
```suggestion
    return alibi_bias

```

add newline

File: examples/llm/src/modules/gpt_blocks.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        key_padding_mask: Optional[torch.ByteTensor] = None,
+        attn_mask: Optional[torch.Tensor] = None,
+    ) -> torch.Tensor:
+        a = self.ln_1(x)
+        b, _ = self.causal_attn(a, key_padding_mask, attn_mask)
+        x = x + self.resid_attn_dropout(b)
+        m = self.ln_2(x)
+        n = self.mlp(m)
+        x = x + self.resid_mlp_dropout(n)
+        return x
```
Comment (vchiley) [inline examples/llm/src/modules/gpt_blocks.py:59]:
```suggestion
        return x

```

add newline

**Review by vchiley — COMMENTED** [review]
missing a few newlines

File: examples/llm/src/modules/__init__.py (hunk 1 above)
Comment (bcui19) [reply examples/llm/src/modules/__init__.py]: rip sorry, ty for pointing it out!

**Comment by abhi-mosaic** [comment]
Lets schedule this for after #179 to avoid merge conflicts

**Comment by bcui19** [comment]
Sounds good. I can merge after #179 lands

**Comment by abhi-mosaic** [comment]
Could I suggest a structure like this? Also `modules` and `models` are really close, maybe we should switch to `layers` or something like that?

```
src/
  layers/
    attention.py
    moe.py # future
  models/   ... from #179 ...
    hf/
      ... from #179 ...
    mosaic_gpt/
      model.py  # keep it single file for now, similar to HF style
    momo1/  # future
      model.py
```

**Comment by abhi-mosaic** [comment]
179 has been merged, safe to rebase now

**Pull request closed without merging**